lang/java/reef-checkpoint/src/main/java/org/apache/reef/io/checkpoint/RandomNameCNS.java

```diff
@@ -26,20 +26,24 @@
 import javax.inject.Inject;
 
 /**
- * Simple naming service that generates a random checkpoint name.
+ * A naming service that generates a random checkpoint name by appending a random alphanumeric string (suffix)
+ * of a given length to a user-supplied prefix string.
  */
 public class RandomNameCNS implements CheckpointNamingService {
 
   private final String prefix;
+  private final int lengthOfRandomSuffix;
 
   @Inject
-  public RandomNameCNS(@Parameter(PREFIX.class) final String prefix) {
+  public RandomNameCNS(@Parameter(PREFIX.class) final String prefix,
```

Comment:
>  Maybe keep the old constructor and use a default for `lengthOfRandomSuffix`

Sounds reasonable. Do you know if I can in code extract the default value from the named parameter `LengthOfRandomSuffix` defined in the same file ?